source/agora/node/BlockStorage.d

```diff
@@ -93,6 +93,21 @@ public interface IBlockStorage
 
     public bool saveBlock (const ref Block block);
 
+    /***************************************************************************
+
+        Update block in the storage with updated signature and bitfield for
+        validator signers.
+
+        Params:
+            block = `Block` to save
+
+        Returns:
+            Returns true if success, otherwise returns false.
+
+    ***************************************************************************/
+
+    public bool updateBlockMultiSig (const ref Block block);
```

Comment:
I wouldn't call this `updateBlockMultiSig`. It's updating the entire block, and not just the bitfield & signature. If it was just updating that pair then those would be the parameters.

Reply:
It is only intended to be used for updating the fixed size fields of the signature and validator bitmask ( Fortuantely we know the max number of signers possible before we externalise so the bitField will never increase in size). This function will abort if a different size for the block is provided. So it can not be used to add or update transactions for example.

Reply:
The problem is now the `BlockStorage` itself has block validation in it. For example comparing the merkle roots. The block storage should just be a plain dumb (optimized) data store, it should not have any validation in it. That belongs elsewhere like in the Ledger for example.

Reply:
Anyway as I mentioned before, since we're on a tight deadline you can leave this as it is now. But we'll have to fix this later.

In particular the BlockStorage throwing exceptions because of a wrong merkle root doesn't really make sense. That's a coding error, not a recoverable exception. The ledger should never attempt to update a block if the merkle root is suddenly different.

Reply:
I raised new issue https://github.com/bpfkorea/agora/issues/1439